### PR TITLE
Unable to delete Infinispan after cluster upgrade from 1.1.x to 2.0.x

### PR DESCRIPTION
**Pull request by dmvolod**
Implementing #549 for version 2.0.x

### CODE REVIEW

**Comment by dmvolod** [comment]
@rigazilla should we forwardport this PR to the master? 
It depends of the upgrade workflow definition for the 2.0.x and further versions. Which strategy you select for upgrades? 
https://docs.openshift.com/container-platform/4.4/operators/understanding_olm/olm-understanding-olm.html

File: pkg/controller/infinispan/util/kubernetes.go [hunk 1]
```diff
@@ -256,9 +256,9 @@ func (k Kubernetes) HasServiceCAsCRDResource() bool {
 	return status >= 200 && status < 299
 }
 
-// GetK8sResources returns the list of the type list of resources associated to the cluster
-func (k Kubernetes) GetK8sResources(name, namespace string, list runtime.Object) error {
-	labelSelector := labels.SelectorFromSet(common.LabelsResource(name, ""))
+// ResourcesTypeList returns a typed list of resource associated with the cluster
+func (k Kubernetes) ResourcesTypeList(name, namespace, resourceType string, list runtime.Object) error {
```
Comment (rigazilla) [inline pkg/controller/infinispan/util/kubernetes.go:260]: maybe `ResourcesList`is enough?

**Review by rigazilla — COMMENTED** [review]
> @rigazilla should we forwardport this PR to the master?
> It depends of the upgrade workflow definition for the 2.0.x and further versions. Which strategy you are selecting for upgrades?
> https://docs.openshift.com/container-platform/4.4/operators/understanding_olm/olm-understanding-olm.html

Channels are organized in z-stream: https://docs.openshift.com/container-platform/4.4/operators/understanding_olm/olm-understanding-olm.html#olm-upgrades-z-stream_olm-understanding-olm
anyway, if there are no drawbacks, I would say to port this PR also on master, so we don't need to set assumption in the code on how OLM channels looks like

**Comment by dmvolod** [comment]
Thanks for review, @rigazilla 
Fixed and will port it to master

**Review by rigazilla — APPROVED** [review]

**Comment by rigazilla** [comment]
merged, thanks @dmvolod !

**Pull request merged**